cli/foundation-models/system/inference/image-text-embeddings/prepare_data.py

```diff
@@ -0,0 +1,234 @@
+
+
+def prepare_data_for_online_inference(dataset_dir: str) -> None:
+    """Prepare request json files for online inference.
+
+    :param dataset_dir: dataset directory
+    :type dataset_dir: str
+    """
+    sample_image_1 = os.path.join(dataset_dir, "milk_bottle", "99.jpg")
+    sample_image_2 = os.path.join(dataset_dir, "can", "1.jpg")
+
+    # Generate sample request for image embeddings
+    image_request_json = {
+        "input_data": {
+            "columns": ["image", "text"],
+            "index": [0, 1],
+            "data": [
+                [
+                    base64.encodebytes(read_image(sample_image_1)).decode("utf-8"),
+                    "",
+                ],  # the "text" column should contain empty string
+                [base64.encodebytes(read_image(sample_image_2)).decode("utf-8"), ""],
+            ],
+        }
+    }
+
+    request_file_name = os.path.join(dataset_dir, "image_request_data.json")
+
+    with open(request_file_name, "w") as request_file:
+        json.dump(image_request_json, request_file)
+
+    # Generate sample request for text embeddings
+    text_request_json = {
+        "input_data": {
+            "columns": ["image", "text"],
+            "index": [0, 1],
+            "data": [
+                ["", "text sample 1"],  # the "text" column should contain empty string
+                ["", "text sample 2"],
+            ],
+        }
+    }
+
+    request_file_name = os.path.join(dataset_dir, "text_request_data.json")
+
+    with open(request_file_name, "w") as request_file:
+        json.dump(text_request_json, request_file)
+
+    # Generate sample request for image and text embeddings
+    image_text_request_json = {
+        "input_data": {
+            "columns": ["image", "text"],
+            "index": [0, 1],
+            "data": [
+                [
+                    base64.encodebytes(read_image(sample_image_1)).decode("utf-8"),
+                    "text sample 1",
+                ],  # all rows should have both images and text
+                [
+                    base64.encodebytes(read_image(sample_image_2)).decode("utf-8"),
+                    "text sample 2",
+                ],
+            ],
+        }
+    }
+
+    request_file_name = os.path.join(dataset_dir, "image_text_request_data.json")
+
+    with open(request_file_name, "w") as request_file:
+        json.dump(image_text_request_json, request_file)
+
+
+def prepare_data_for_batch_inference(dataset_dir: str) -> None:
+    """Prepare image folder and csv files for batch inference.
+
+    This function will move all images to a single image folder and also create folders of csv
+    files. Each folder will have csv files that contain images in base64 format, text samples, or both.
+    :param dataset_dir: dataset directory
+    :type dataset_dir: str
+    """
+    batch_input_file = "batch_input.csv"
+    # Generate batch input for image embeddings
+    image_list = []
+
+    dir_names = []
```

Comment:
Would it be better to use the folder name as the text column instead of using a random text message